### PR TITLE
BroomCat: upgrade to apiv2

### PR DESCRIPTION
**Pull request by kazzarin**
- switched to apiv2 and added additional rate limit handling
  - it will now retry after the wait period is over
- added the ability to sort by "date added" instead of by popularity

### CODE REVIEW

File: src/queries/BroomCat.js [hunk 1]
```diff
@@ -5,7 +5,22 @@
...
 	animeOption.value = "ANIME";
 	mangaOption.value = "MANGA";
-	createCheckbox(miscOptions,"restrictToList");
+	let sortSelect = create("select","#sortSelect",false,miscOptions);
+	let popOption = create("option",false,"Popularity",sortSelect);
+	let idOption = create("option",false,"Date Added",sortSelect);
+	popOption.value = "POPULARITY_DESC";
+	idOption.value = "ID_DESC";
+	let listRestrict = createCheckbox(miscOptions,"restrictToList");
+	listRestrict.onchange = function(){
```
Comment (hohMiyazawa) [inline src/queries/BroomCat.js:14]: Why disable sorting here?

Comment (kazzarin) [reply src/queries/BroomCat.js:14]: There's two different queries and I was only focused on adding sort for the full media query. (They use different sort enums `MediaSort` vs `MediaListSort`)

I can look into adding (more) sort options for both in either a merged select that toggles available options or two different selects based on the state of the checkbox.

Comment (hohMiyazawa) [reply src/queries/BroomCat.js:14]: Ah, forgot that those are different sort enums.

**Comment by hohMiyazawa** [comment]
Couple of more options: https://github.com/hohMiyazawa/Automail/commit/bb6b6298fb25fae607caf4e34b30ad442f93b696

**Pull request merged**